lib/connection_pool.rb

```diff
@@ -53,21 +53,14 @@ def initialize(options = {}, &block)
   end
 
   def with(options = {})
-    # Connections can become corrupted via Timeout::Error.  Discard
-    # any connection whose usage after checkout does not finish as expected.
-    # See #67
-    success = false
-    conn = checkout(options)
-    begin
-      result = yield conn
-      success = true # means the connection wasn't interrupted
-      result
-    ensure
-      if success
-        # everything is roses, we can safely check the connection back in
+    Thread.handle_interrupt(StandardError => :never) do
```

Comment:
stub `checkout` to sleep a while and then "do something", and assert that that "something" got "done", and also that `checkin` got called. you'll want to call `with` inside a `timeout` obviously